File: frontend/src/types/responce/teamFatigueRes.ts
import type FatigueLog from "../fatigueLog";
import type Team from "../team";
import type User from "../user";

export default interface TeamFatigueRes {
  team_user: User[];
  team_data: Team;
  fatigue_logs: {
    [user_id: string]: FatigueLog[];
  };
}

File: desktop/desktop/Models/DTOs/Responses/teamFatigueRes.cs
using System.Collections.Generic;
using System.Text.Json.Serialization;

public record TeamFatigueRes(
    [property: JsonPropertyName("team_user")]
    List<User> TeamUser,

    [property: JsonPropertyName("team_data")]
    Team TeamData,

    [property: JsonPropertyName("fatigue_logs")]
    Dictionary<string, List<FatigueLog>> FatigueLogs
);